src/tr/merlinscans/src/eu/kanade/tachiyomi/extension/tr/merlinscans/MerlinScans.kt

```diff
@@ -1,12 +1,431 @@
+            setUrlWithoutDomain(normalizeUrl(seriesLink))
+            title = element.select("h3.series-title a").text()
+            thumbnail_url = element.select("div.series-cover-thumbnail img").attr("src")
+        }
+    }
+
+    override fun latestUpdatesNextPageSelector(): String? = null
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        if (query.isNotEmpty()) {
+            val url = "$baseUrl/ajax/search-preview.php".toHttpUrl().newBuilder()
+                .addQueryParameter("q", query)
+                .build()
+            return GET(url, headers)
+        }
+
+        val url = "$baseUrl/all-series.php".toHttpUrl().newBuilder()
+            .addQueryParameter("page", page.toString())
+
+        var sortApplied = false
+
+        filters.forEach { filter ->
+            when (filter) {
+                is TypeFilter -> {
+                    if (filter.state != 0) {
+                        url.addQueryParameter("type", filter.toUriPart())
+                    }
+                }
+
+                is CategoryFilter -> {
+                    if (filter.state != 0) {
+                        url.addQueryParameter("category", filter.toUriPart())
+                    }
+                }
+
+                is StatusFilter -> {
+                    if (filter.state != 0) {
+                        url.addQueryParameter("status", filter.toUriPart())
+                    }
+                }
+
+                is SortFilter -> {
+                    if (filter.state != 0) {
+                        url.addQueryParameter("sort", filter.toUriPart())
+                        sortApplied = true
+                    }
+                }
+
+                else -> {}
+            }
+        }
+
+        if (!sortApplied) {
+            url.addQueryParameter("sort", "views")
+        }
+
+        return GET(url.build(), headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val requestUrl = response.request.url.toString()
+
+        return if (requestUrl.contains("search-preview.php")) {
+            parseSearchJson(response)
+        } else {
+            super.searchMangaParse(response)
+        }
+    }
+
+    private fun parseSearchJson(response: Response): MangasPage {
+        val searchResponse: SearchResponse = response.parseAs()
+
+        val mangas = if (searchResponse.success) {
+            searchResponse.results.map { result ->
+                SManga.create().apply {
+                    title = result.title
+                    thumbnail_url = result.coverImage
+                    setUrlWithoutDomain(normalizeUrl(result.url))
+
+                    if (result.categories.isNotEmpty()) {
+                        genre = result.categories.split(", ")
+                            .distinct()
+                            .take(10)
+                            .joinToString(", ")
```

Comment:
`, ` is the default and can be omitted. Might be better to keep it in this context, though.